main.go

```diff
@@ -356,20 +363,22 @@ func processSource(nsCfg config.NamespaceConfig, t tail.Follower, parser *gonx.P
 			}
 		}
 
+		notCounterValues := relabeling.StripOnlyCounterValues(labelValues, relabelings)
+
 		metrics.countTotal.WithLabelValues(labelValues...).Inc()
 
 		if bytes, ok := floatFromFields(fields, "body_bytes_sent"); ok {
-			metrics.bytesTotal.WithLabelValues(labelValues...).Add(bytes)
+			metrics.bytesTotal.WithLabelValues(notCounterValues...).Add(bytes)
 		}
 
 		if upstreamTime, ok := floatFromFields(fields, "upstream_response_time"); ok {
-			metrics.upstreamSeconds.WithLabelValues(labelValues...).Observe(upstreamTime)
-			metrics.upstreamSecondsHist.WithLabelValues(labelValues...).Observe(upstreamTime)
+			metrics.upstreamSeconds.WithLabelValues(notCounterValues...).Observe(upstreamTime)
+			metrics.upstreamSecondsHist.WithLabelValues(notCounterValues...).Observe(upstreamTime)
 		}
 
 		if responseTime, ok := floatFromFields(fields, "request_time"); ok {
-			metrics.responseSeconds.WithLabelValues(labelValues...).Observe(responseTime)
-			metrics.responseSecondsHist.WithLabelValues(labelValues...).Observe(responseTime)
+			metrics.responseSeconds.WithLabelValues(notCounterValues...).Observe(responseTime)
+			metrics.responseSecondsHist.WithLabelValues(notCounterValues...).Observe(responseTime)
```

Comment:
Could you maybe also add a test case for this new behaviour? In the `features/` directory in this repo there are some [Behave](https://behave.readthedocs.io/en/latest/) tests with example configuration files, example input log files and expected outputs. This would help assert that this feature does not break in subsequent releases (as I'm probably going to forget testing this feature manually).

Reply:
See the newly added `.feature` file. Hope this goes into the right direction.